dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/IndiciesInfo.java

```diff
@@ -0,0 +1,233 @@
+        this.indiciesNames.put(IndexType.SITE_SEARCH, builder.siteSearch);
+    }
+
+    /**
+     * Return the current LIVE index name
+     * @return
+     */
+    public String getLive() {
+        return this.indiciesNames.get(IndexType.LIVE);
+    }
+
+    /**
+     * Return the current WORKING index name
+     * @return
+     */
+    public String getWorking() {
+        return this.indiciesNames.get(IndexType.WORKING);
+    }
+
+    /**
+     * Return the current REINDEX LIVE index name
+     * @return
+     */
+    public String getReindexLive() {
+        return this.indiciesNames.get(IndexType.REINDEX_LIVE);
+    }
+
+    /**
+     * Return the current REINDEX WORKING index name
+     * @return
+     */
+    public String getReindexWorking() {
+        return this.indiciesNames.get(IndexType.REINDEX_WORKING);
+    }
+
+    /**
+     * Return the current SITE SEARCH index name
+     * @return
+     */
+    public String getSiteSearch() {
+        return this.indiciesNames.get(IndexType.SITE_SEARCH);
+    }
+
+    /**
+     * Return the timestamp from a index name, a index name has the follow sintax:
+     *
+     * <b>cluster_<CLUSTER_ID>.<INDEX_NAME_PREFIX>_<TIME_STAMP></b>
+     * @return
+     */
+    public long getIndexTimeStamp(final IndexType indexType) {
+        Date startTime;
+        try {
+            final String indexName = indiciesNames.get(indexType);
+            final String indexTimestamp = indexName.substring(indexName.lastIndexOf("_") + 1);
+            startTime = timestampFormatter.parse(indexTimestamp);
+
+            return System.currentTimeMillis() - startTime.getTime();
+        } catch (ParseException e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    /**
+     * Create a new indicies name for the type pass in indiciesType
+     *
+     * @param indiciesType types for what you want to create a new name
+     * @return timestamp for the newly indicies, to get the newly indices name use one of the follow methods:
+     * {@link IndiciesInfo#getLive()}, {@link IndiciesInfo#getWorking()}, {@link IndiciesInfo#getReindexLive()},
+     * {@link IndiciesInfo#getReindexWorking()} {@link IndiciesInfo#getSiteSearch()}
+     */
+    public String createNewIndiciesName(final IndexType... indiciesType) {
+        final String timeStamp = timestampFormatter.format(new Date());
+
+        for (final IndexType indexType : indiciesType) {
+            final String indexName = String.format(
+                    INDEX_NAME_PATTERN,
+                    ClusterFactory.getClusterId(),
+                    indexType.getPrefix(),
+                    timeStamp);
+
+            this.indiciesNames.put(indexType, indexName);
+        }
+
+        return timeStamp;
+    }
+
+    public Map<IndexType, String> asMap() {
+        Map<IndexType, String> actives = new HashMap<>();
+        for (IndexType type : IndexType.values()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'type' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=3712629)